web-api-db-ctrl/src/postgres/extern_migrations.rs

```diff
@@ -0,0 +1,123 @@
+// Copyright 2023 Xayn AG
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as
+// published by the Free Software Foundation, version 3.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+use std::future::Future;
+
+use anyhow::bail;
+use async_trait::async_trait;
+use derive_more::{From, Into};
+use sqlx::{Postgres, Transaction, Type};
+
+use crate::Error;
+
+/// Allows running migration like "once per tenant" tasks.
+#[async_trait(?Send)]
+pub(crate) trait ExternalMigrator {
+    async fn run_migration_if_needed(
+        &mut self,
+        name: &str,
+        exec: impl Future<Output = Result<(), Error>>,
+    ) -> Result<(), Error>;
+}
+
+#[derive(From, Into)]
+pub(crate) struct PgExternalMigrator {
+    tx: Transaction<'static, Postgres>,
+}
+
+impl PgExternalMigrator {
+    fn tx(&mut self) -> &mut Transaction<'static, Postgres> {
+        &mut self.tx
+    }
+}
```

Comment:
:art: we could just inline `&mut self.tx` at the three places where this function is called.
```suggestion
```

Reply:
okay, through it was convenient to have as during development the type of `tx` was changed multiple times